research/2019-12-16_searching_with_the_calm_api.md

```diff
@@ -0,0 +1,241 @@
+# Searching with the CALM API
+
+### 16 December 2019, AWLC
+
+This has some notes on using the CALM API, and an example script for searching all the records in the CALM catalogue.
+
+
+
+## Motivation
+
+As part of the born-digital content migration, I had a collection of directories with accession-like numbers, and I wanted to get a list of all the accession numbers in CALM, so I could match the directories to accession numbers:
+
+    AAUICW_2185 ~> AAUICW/2185 ?
+    AAUNAM_2213 ~> AAUNAM/2213 ?
+    AAUNAM_2254 ~> AAUNAM/2254 ?
+
+Exporting a list of accession numbers from the CALM GUI would have been tricky (I don't have access to CALM on my Mac), so I tried using the CALM API to export the data.
+
+<details>
+  <summary>What is an accession/accession number?</summary>
+
+  > In library science, an *accession* is all the material that gets collected at one time from a single source.
+  > The *accession identifier* is the unique identifier given to that batch of material.
+  >
+  > At Wellcome, the accession identifier is made of two parts: the *accession number* (sequential, currently up to 2551) and a *collection reference* (such as WT, PPMTW, SANCT).
+  > The two are written together, separated by a slash.
+  > Examples:
+  >
+  > The accession number is globally unique: if there's an accession `WT/123`, you won't see `123` with any other collection reference.
+</details>
```

Comment:
What syntax is this?

Reply:
Nice note though!

Reply:
This is the HTML5 `<details>` tag: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

It gives you a widget that allows you to selectively show/hide information. I expect at least some people reading this will know what an accession (number) is, so they can skip this bit – but if you don’t, you won’t be lost.

Reply:
This is the HTML5 `<details>` tag: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

It gives you a widget that allows you to selectively show/hide information. I expect at least some people reading this will know what an accession (number) is, so they can skip this bit – but if you don’t, you won’t be lost.